cubi_tk/sodar/ingest.py

```diff
@@ -0,0 +1,297 @@
+        )
+        parser.add_argument(
+            "-r",
+            "--recursive",
+            default=False,
+            action="store_true",
+            help="Recursively match files in subdirectories. Creates iRODS sub-collections to match directory structure.",
+        )
+        parser.add_argument(
+            "-s",
+            "--sync",
+            default=False,
+            action="store_true",
+            help="Skip upload of files already present in remote collection.",
+        )
+        parser.add_argument(
+            "-e",
+            "--exclude",
+            nargs="+",
+            default="",
+            type=str,
+            help="Exclude files by defining one or multiple glob-style patterns.",
+        )
+        parser.add_argument(
+            "-K",
+            "--remote-checksums",
+            default=False,
+            action="store_true",
+            help="Trigger checksum computation on the iRODS side.",
+        )
+        parser.add_argument(
+            "-y",
+            "--yes",
+            default=False,
+            action="store_true",
+            help="Don't ask for permission.",
+        )
+        parser.add_argument(
+            "--collection",
+            type=str,
+            help="Target iRODS collection. Skips manual selection input.",
+        )
+        parser.add_argument(
+            "sources", help="One or multiple files/directories to ingest.", nargs="+"
+        )
+        parser.add_argument("destination", help="UUID or iRODS path of SODAR landing zone.")
+
+    @classmethod
+    def run(
+        cls, args, _parser: argparse.ArgumentParser, _subparser: argparse.ArgumentParser
+    ) -> typing.Optional[int]:
+        """Entry point into the command."""
+        return cls(args).execute()
+
+    def execute(self):
+        """Execute ingest."""
+        # Retrieve iRODS path if destination is UUID
+        if is_uuid(self.args.destination):
+            try:
+                lz_info = api.landingzone.retrieve(
+                    sodar_url=self.args.sodar_url,
+                    sodar_api_token=self.args.sodar_api_token,
+                    landingzone_uuid=self.args.destination,
+                )
+            except Exception as e:  # pragma: no cover
+                logger.error("Failed to retrieve landing zone information.")
+                logger.exception(e)
+                sys.exit(1)
+
+            # TODO: Replace with status_locked check once implemented in sodar_cli
+            if lz_info.status in ["ACTIVE", "FAILED"]:
+                self.lz_irods_path = lz_info.irods_path
+                logger.info(f"Target iRods path: {self.lz_irods_path}")
+            else:
+                logger.error("Target landing zone is not ACTIVE.")
+                sys.exit(1)
+        else:
+            self.lz_irods_path = self.args.destination  # pragma: no cover
+
+        # Build file list
+        source_paths = self.build_file_list()
```

Comment:
I think it would be nice if all ingest/itransfer functions can use the same function namesin places like this.
However, this will likely need changes across cubi-tk and we need to agree on a standard

Reply:
Maybe, even though not much is gained from this if you ask me. I would however discuss this at a later time.

Reply:
Yeah this needs to be discussed overall (and probably also needs to be done on all functions at the same time)